### PR TITLE
Enable xnack code gen only if it is enabled in the driver

### PR DESCRIPTION
**Pull request by crystalJinamd**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit c38c1c23eb4f21e004a3af1a0f71abe4c18c887a
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6cc0a035259663610287d8d99262505e78e01db2</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev f1bf5feb06f0d3c3e64e960797a35cee3eae4aa4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 413996e376673364c2c03e5fef2744753c867171</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

File: lgc/state/LgcContext.cpp [hunk 1]
```diff
@@ -234,6 +236,8 @@ LgcContext *LgcContext::Create(LLVMContext &context, StringRef gpuName, unsigned
     return nullptr;
   }
 
+  builderContext->m_targetInfo->getGpuProperty().pageMigrationEnabled = pageMigrationEnabled;
```
Comment (amdrexu) [inline lgc/state/LgcContext.cpp:239]: 1. If `pageMigrationEnabled `depends on GFX IP, we have to set it to true or false according to GFX IP value. And this is done by `setTargetInfoXXX`.
2. If `pageMigrationEnabled` depends on driver settings, we have to add a pipeline option in vkgcDefs and in LGC pipeline options. And this is passed from driver to LLPC front-end and finally to LGC.
Anyway, we always avoid adding a new parameter in creation function.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8669d1b258d7b17963ad0b53149d7794e4ddc16f
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6cc0a035259663610287d8d99262505e78e01db2</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev f1bf5feb06f0d3c3e64e960797a35cee3eae4aa4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 413996e376673364c2c03e5fef2744753c867171</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

File: lgc/state/LgcContext.cpp (hunk 1 above)
Comment (crystalJinamd) [reply lgc/state/LgcContext.cpp]: Thanks, your suggestion is more reasonable.
It is fixed in the latest commit, please help to review it, thanks !

**Comment by amdvlk-admin** [comment]
### Test summary for commit ddb57c8ef3264dd00a162d5f1fc84b572f3a11c7
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6cc0a035259663610287d8d99262505e78e01db2</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev f1bf5feb06f0d3c3e64e960797a35cee3eae4aa4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 413996e376673364c2c03e5fef2744753c867171</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — COMMENTED** [review]
LGTM. Don't forget to make a Vulkan driver change to enable this option.

**Comment by crystalJinamd** [comment]
> LGTM. Don't forget to make a Vulkan driver change to enable this option.

Yes, I will make a vulkan change after this one being merged.

**Comment by JaxLinAMD** [comment]
on one approve?

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2d324939eac3c644878f5fd9715c6d1577942bc9
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 6cc0a035259663610287d8d99262505e78e01db2</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev 089c9244e6baa5ed42b1342b391c7c6aec521a2f</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev ad9f69a6cf94a03817d1d40bacf1920f31e3c727</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Pull request merged**